enterprise/coderd/prebuilds/reconcile.go

```diff
@@ -243,6 +243,11 @@ func (c *StoreReconciler) ReconcileAll(ctx context.Context) error {
 		var eg errgroup.Group
 		// Reconcile presets in parallel. Each preset in its own goroutine.
 		for _, preset := range snapshot.Presets {
+			if snapshot.IsHardLimited(preset.ID) {
```

Comment:
This is inconsistent with how we perform the other defensive measures.
The better place for this would be `ReconcilePreset`.

Reply:
Reimplemented it here: https://github.com/coder/coder/pull/17697/commits/0fdd096951a7a9219f9be05f4a8b827766dd59e9

Now it's more consistent.